models/models_methods.go

```diff
@@ -299,14 +299,41 @@ func (workflow *WorkflowModel) Validate() ([]string, error) {
 				warnings = append(warnings, fmt.Sprintf("invalid step: duplicated input found: (%s)", key))
 			}
 			stepInputMap[key] = true
-		}
 
+			opts, err := input.GetOptions()
+			if err != nil {
+				return warnings, err
+			}
+			if s := opts.IsSensitive; (s != nil && *s) && (opts.IsExpand != nil && !*opts.IsExpand) {
```

Comment:
isSensitive := opts.IsSensitive
if isSensitive == nil {
  isSensitive = defaultIsSensitive
}

(same for IsExpand)

if isSensitive && ! IsExpand {
...
}